pkg/asset/installconfig/azure/validation_test.go

```diff
@@ -324,6 +344,16 @@ func TestAzureInstallConfigValidation(t *testing.T) {
 			edits:    editFunctions{validInstanceTypes},
 			errorMsg: "",
 		},
+		{
+			name:     "Valid Arm64 instance types",
+			edits:    editFunctions{validArm64InstanceTypes},
+			errorMsg: "",
+		},
+		{
+			name:     "Invalid arch instance types",
+			edits:    editFunctions{invalidArchInstanceTypes},
+			errorMsg: `\[controlPlane.platform.azure.type: Invalid value: "Standard_D8ps_v5": instance type architecture does not match install config architecture amd64, compute\[0\].platform.azure.type: Invalid value: "Standard_D4ps_v5": instance type architecture does not match install config architecture amd64, platform.azure.defaultMachinePlatform.type: Invalid value: "Standard_D4ps_v5": instance type architecture does not match install config architecture amd64\]`,
+		},
```

Comment:
It would be good to have a test to cover the case when the `defaultMachinePlatform` is used for either controlPlane or compute so your last check in `validateInstanceTypes` is exercised.

Reply:
this test covers that case too - notice the last error `platform.azure.defaultMachinePlatform.type: Invalid value: "Standard_D4ps_v5": instance type architecture does not match install config architecture amd64`

Reply:
My comment was a result of my misunderstanding of what your code was doing.

Reply:
so are we all good on this then ?